compiler/ast/ast_types.nim

```diff
@@ -1535,8 +1533,8 @@ type
         discard
 
   TNode*{.final, acyclic.} = object # on a 32bit machine, this takes 32 bytes
-    id*: NodeId
     typ*: PType
+    id*: NodeId
```

Comment:
```suggestion
    id*: NodeId  # placed after `typ` field to save space due to field alignment
```

soft suggestion, just in case someone gets excited and wants to reorder the fields -- aka future me.